event_model/__init__.py

```diff
@@ -1045,6 +1045,73 @@ def unpack_datum_page(datum_page):
         yield datum
 
 
+def rechunk_event_pages(event_pages, chunk_size):
+    remainder = chunk_size
+    chunk_list = []
+
+    for page in event_pages:
+        new_chunks = page_chunks(page, chunk_size, remainder)
+        for chunk in new_chunks:
+            remainder -= len(chunk['uid'])
+            chunk_list.append(chunk)
+            if remainder == 0:
```

Comment:
Should this be `<=`? (What happens if `page_size` isn't divisible by `len(chunk['uid'])`. A good test case to add.

Reply:
I think we want ==, but I'm not totally sure. I don't think remainder will ever be negative, even if page size isn't evenly divisible by chunk size. That's what I intended at least. Good idea, ill write a test case for this.